sentry_sdk/profiler.py

```diff
@@ -439,6 +473,14 @@ def slice_profile(self, start_ns, stop_ns):
 
     def make_sampler(self):
         # type: () -> Callable[..., None]
+        cwd = os.getcwd()
+
+        # In Python3+, we can use the `nonlocal` keyword to rebind the value,
+        # but this is not possible in Python2. To get around this, we wrap
+        # the value in a list to allow updating this value each sample.
+        last_sample = [
```

Comment:
So the sample buffer is going away in the next PR because keeping all these samples around was very memory intensive for low memory applications with lots of short profiles. Hence I opted to keep this here.